File: dotnet_webapi_mail/controllers/EmailController.cs

using dotnet_webapi_mail.service;
using Microsoft.AspNetCore.Mvc;
using System.Threading.Tasks;

namespace dotnet_webapi_mail.controllers;

[ApiController]
[Route("api/[controller]")]
public class EmailController : ControllerBase
{
    private readonly EmailService _emailService;
    public EmailController(EmailService emailService) => _emailService = emailService;

    [HttpPost("send")]
    public async Task<IActionResult> SendEmail(string to, string subject, string body)
    {
        await _emailService.SendEmailAsync(to, subject, body);
        return Ok("Email sent successfully!");
    }
}

File: demo/src/main/java/com/example/demo/controller/EmailController.java
package com.example.demo.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import com.example.demo.service.EmailService;

@RestController
@RequestMapping("/api/mail")
public class EmailController {

    @Autowired
    private EmailService emailService;

    @PostMapping("/send")
    public String sendMail(@RequestParam String to, @RequestParam String subject, @RequestParam String body) {
        emailService.sendEmail(to, subject, body);
        return "Email sent successfully!";
    }
}